gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,595 @@
+import logging
+import random
+import numpy
+
+import numpy as np
+from gensim.scripts.glove2word2vec import glove2word2vec
+from gensim.test.utils import get_tmpfile
+from gensim.models import KeyedVectors
+from collections import Counter
+from custom_losses import rank_hinge_loss
+from custom_layers import TopKLayer
+from sklearn.preprocessing import normalize
+from custom_callbacks import ValidationCallback
+from gensim import utils
+
+try:
+    import keras.backend as K
+    from keras import optimizers
+    from keras.losses import hinge
+    from keras.models import Model
+    from keras.layers import Input, Embedding, Dot, Dense, Reshape, Dropout
+    import tensorflow
+    tensorflow.set_random_seed(101010)
```

Comment:
no global seeding, never, because you affect to general interpreter state (i.e. you seed not only your model, you seed all random).
The right way - create random-generator first and seed it (this is about all `*seed` calls)